packages/fast-components-react-msft/src/carousel/carousel.tsx

```diff
@@ -110,15 +127,16 @@ class Carousel extends Foundation<
      */
     protected generateClassNames(): string {
         let className: string = get(this.props, "managedClasses.carousel", "");
-        let theme: string = "";
 
         if (this.getSlideTheme()) {
-            theme =
-                this.getSlideTheme() === SlideTheme.light
-                    ? get(this.props, "managedClasses.carousel__themeLight", "")
-                    : get(this.props, "managedClasses.carousel__themeDark", "");
+            className += this.assignSlideThemeClassName();
+        }
 
-            className += ` ${theme}`;
+        if (
+            this.slideTransitionDirection === SlideTransitionDirection.next ||
+            this.slideTransitionDirection === SlideTransitionDirection.previous
```

Comment:
Do we need to check both of these because "internal" exists? Is it possible we could simplify this and just check for this.slideTransitionDirection?

Reply:
changed